libr/anal/var.c

```diff
@@ -793,9 +793,15 @@ R_API RList *r_anal_var_all_list(RAnal *anal, RAnalFunction *fcn) {
 	if (!list) {
 		return NULL;
 	}
-	r_list_join (list, r_anal_var_list (anal, fcn, R_ANAL_VAR_KIND_REG));
-	r_list_join (list, r_anal_var_list (anal, fcn, R_ANAL_VAR_KIND_BPV));
-	r_list_join (list, r_anal_var_list (anal, fcn, R_ANAL_VAR_KIND_SPV));
+	RList *reg_vars = r_anal_var_list (anal, fcn, R_ANAL_VAR_KIND_REG);
+	RList *bpv_vars = r_anal_var_list (anal, fcn, R_ANAL_VAR_KIND_BPV);
+	RList *spv_vars = r_anal_var_list (anal, fcn, R_ANAL_VAR_KIND_SPV);
+	r_list_join (list, reg_vars);
```

Comment:
you can remove one of those lists if you just do RList *list = reg_vars; in the first line

Reply:
RList *list is not an empty pointer, it is another list declared some lines before, that needs to be merged to reg_vars, bpv_vars, spv_vars...